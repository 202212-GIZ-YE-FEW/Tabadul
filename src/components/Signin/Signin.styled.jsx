import styled from "@emotion/styled";
import Image from "next/image";

export const SigninContainer = styled.div``;

export const Itemscontainer = styled.div`
    display: flex;
    justify-content: center;
    items-align: center;
`;

export const Inputscontainer = styled.div`
    display: flex;
    flex-direction: column;
    padding: 110px;
    @media screen and (max-width: 600px) {
        margin-left: 400px;
    }
    @media screen and (max-width: 420px) {
        margin-left: 700px;
    }
`;

export const Signinheader = styled.h1`
    width: 206px;
    height: 31px;
    font-weight: 700;
    font-size: 36px;
    line-height: 54px;
    text-align: center;
    color: #3c4347;
    margin-left: 180px;
    @media screen and (max-width: 600px) {
        width: 99px;
        height: 26px;
        font-size: 24px;
        margin-left: 300px;
    }
`;

export const Signinform = styled.form``;
export const Containerinput = styled.div`
    display: flex;
    flex-direction: column;
`;

export const Itemsdev = styled.div`
    display: flex;
    flex-direction: column;
    padding: 20px;
    margin-top: 10px;
`;
export const Test = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;
export const Signintitle = styled.label`
    width: 55px;
    height: 30px;
    left: 50px;
    top: 59px;
    font-weight: 400;
    font-size: 20px;
    line-height: 30px;
    color: #000000;
    margin-left: 20px;
`;
export const Signininput = styled.input`
    box-sizing: border-box;
    width: 500px;
    height: 49px;
    left: 37px;
    top: 100px;
    border: 2px solid #33956d;
    border-radius: 15px;
    font-size: 20px;
    margin-top: 10px;
    &::placeholder {
        font-weight: 400;
        font-size: 16px;
        line-height: 30px;
        color: #cacaca;
        padding-left: 20px;
    }
    @media screen and (max-width: 600px) {
        width: 364px;
        height: 58.77;
    }
`;

export const Checklabel = styled.label`
    width: 298px;
    height: 32px;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;
    text-align: center;
    color: #888c95;
    margin-right: 300px;
    @media screen and (max-width: 600px) {
        margin-left: 120px;
    }
`;
export const Checkinput = styled.input`
    box-sizing: border-box;
    width: 19px;
    height: 19px;
    left: 46px;
    top: 276px;
    border: 1px solid #4e5769;
    border-radius: 6px;
`;

export const Signinbutton = styled.button`
    width: 502px;
    height: 48px;
    background: #33956d;
    border-radius: 20px;
    margin-top: 40px;
    @media screen and (max-width: 600px) {
        width: 271px;
        height: 49px;
    }
`;
export const Buttonspan = styled.span`
    width: 72px;
    height: 36px;
    font-weight: 700;
    font-size: 24px;
    line-height: 36px;
    text-align: center;
    color: #ffffff;
    @media screen and (max-width: 600px) {
        width: 60px;
        height: 30px;
        font-size: 20px;
    }
`;

export const Paraghraph = styled.p`
    width: 327px;
    height: 28px;
    left: 123px;
    top: 403px;
    font-weight: 400;
    font-size: 16px;
    line-height: 20px;
    text-align: center;
    color: #888c95;
    @media screen and (max-width: 600px) {
        font-size: 20px;
    }
`;
export const Linkparagraph = styled.a`
width: 327px;
height: 28px;
font-weight: 400;
font-size: 16px;
line-height: 20px
text-decoration-line: underline;
color: #4E5769;
@media screen and (max-width: 600px) {
    font-size:20px    
}
`;

export const ChangePass = styled.a`
    width: 327px;
    height: 28px;
    font-weight: 400;
    font-size: 16px;
    line-height: 20px;
    text-align: center;
    text-decoration-line: underline;
    color: #4e5769;
    @media screen and (max-width: 600px) {
        font-size: 20px;
    }
`;
export const ParaghraphConatainer = styled.div`
    display: flex;
    flex-direction: column;
    items-align: center;
    padding: 20px;
`;

export const Signinwith = styled.p`
    width: 156px;
    height: 31px;
    font-weight: 700;
    font-size: 16px;
    line-height: 20px;
    text-align: center;
    color: #3c4347;
    @media screen and (max-width: 600px) {
        font-size: 20px;
    }
`;
export const Imgsdev = styled.div`
    display: flex;
`;

export const FaceBookImg = styled(Image)`
    padding: 10px;
    bottom: 0;
`;

export const InstagramImg = styled(Image)`
    padding: 10px;
    bottom: 0;
`;

export const LinedinImg = styled(Image)`
    padding: 10px;
    bottom: 0;
`;

export const ContainerImg = styled.div`
    @media screen and (max-width: 1300px) {
        display: none;
    }
`;
export const Signinimage = styled(Image)``;
