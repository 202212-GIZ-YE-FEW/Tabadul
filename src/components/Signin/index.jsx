import {
    browserSessionPersistence,
    setPersistence,
    signInWithEmailAndPassword,
} from "firebase/auth";
import { useRouter } from "next/router";
import { withTranslation } from "next-i18next";
import React, { useState } from "react";

import { auth } from "@/utils/firebase";

import {
    Buttonspan,
    ChangePass,
    Checkinput,
    Checklabel,
    ContainerImg,
    Containerinput,
    Erorrsmsg,
    FaceBookImg,
    Imgsdev,
    Inputscontainer,
    InstagramImg,
    Itemscontainer,
    Itemsdev,
    LinedinImg,
    Linkparagraph,
    Loadding,
    Loadinglabel,
    Paraghraph,
    ParaghraphConatainer,
    Signinbutton,
    SigninContainer,
    Signinform,
    Signinheader,
    Signinimage,
    Signininput,
    Signintitle,
    Signinwith,
    Test,
} from "./Signin.styled";

function Signin({ t }) {
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
    const [emailError, setEmailError] = useState("");
    const [passError, setPassError] = useState("");
    const [loading, setLoading] = useState(false);
    const [keepLoggedIn, setKeepLoggedIn] = useState(false);
    const r = useRouter();
    const signIn = async (e) => {
        e.preventDefault();
        if (!keepLoggedIn) {
            setPersistence(auth, browserSessionPersistence);
        }
        try {
            if (!email) {
                setEmailError("Please enter your email");
            } else {
                setEmailError("");
            }
            if (!password) {
                setPassError("Please enter your password");
            } else {
                setPassError("");
            }
            // await auth.setPersistence(keepMeLoggedIn ? firebase.auth.Auth.Persistence.SESSION : firebase.auth.Auth.Persistence.LOCAL);
            await signInWithEmailAndPassword(auth, email, password);
            r.push("/");
            setLoading(true);
        } catch (error) {
            console.error(error);
            if (email && password) {
                alert("email or password is not vaild");
            }
            setLoading(false);
        }
    };

    return (
        <SigninContainer>
            <Itemscontainer>
                <Inputscontainer>
                    <Signinheader>{t("Login")}</Signinheader>
                    <Signinform onSubmit={signIn}>
                        <Containerinput>
                            <Test>
                                <Itemsdev>
                                    <Signintitle>{t("Email")}</Signintitle>
                                    <Signininput
                                        id='email'
                                        name='email'
                                        placeholder={t("Enteryouremail")}
                                        type='text'
                                        value={email}
                                        onChange={(e) =>
                                            setEmail(e.target.value)
                                        }
                                        // onBlur={handleSubmit}
                                    />
                                    {emailError && (
                                        <Erorrsmsg>{emailError}</Erorrsmsg>
                                    )}
                                </Itemsdev>
                                <Itemsdev>
                                    <Signintitle>{t("Password")}</Signintitle>
                                    <Signininput
                                        placeholder={t("Enteryourpassword")}
                                        type='password'
                                        value={password}
                                        onChange={(e) =>
                                            setPassword(e.target.value)
                                        }
                                        // onBlur={handleSubmit}
                                    />
                                    {passError && (
                                        <Erorrsmsg>{passError}</Erorrsmsg>
                                    )}
                                </Itemsdev>
                                <Checklabel>
                                    <Checkinput
                                        type='checkbox'
                                        checked={keepLoggedIn}
                                        onChange={(event) =>
                                            setKeepLoggedIn(
                                                event.target.checked
                                            )
                                        }
                                    />
                                    {t("keepmeloggedin")}
                                </Checklabel>
                                <Signinbutton
                                    type='submit'
                                    // onClick={handleSubmit}
                                >
                                    <Buttonspan>{t("Loginbutton")}</Buttonspan>
                                </Signinbutton>
                                {loading && (
                                    <Loadding>
                                        <i></i>
                                        <Loadinglabel>Loading...</Loadinglabel>
                                    </Loadding>
                                )}
                            </Test>
                        </Containerinput>
                    </Signinform>
                    <Test>
                        <ParaghraphConatainer>
                            <Paraghraph>
                                {t("Dothaveaccount")}
                                <Linkparagraph href='/Signup'>
                                    {t("Signup")}
                                </Linkparagraph>
                            </Paraghraph>
                            <ChangePass href='/Forgetpassword'>
                                {t("ForgetPassword")}
                            </ChangePass>
                        </ParaghraphConatainer>
                        <Signinwith>{t("Loginwith")}</Signinwith>
                        <Imgsdev>
                            <FaceBookImg
                                src='/images/Facebook.svg'
                                alt='Facebook'
                                width={42.3}
                                height={42.3}
                            />
                            <InstagramImg
                                src='/images/Instagram.svg'
                                alt='Instagram'
                                width={42.3}
                                height={42.3}
                            />
                            <LinedinImg
                                src='/images/Linkedin.svg'
                                alt='Linkedin'
                                width={42.3}
                                height={42.3}
                            />
                        </Imgsdev>
                    </Test>
                </Inputscontainer>
                <ContainerImg>
                    <Signinimage
                        src='/images/Signin.svg'
                        alt='sign in'
                        width={750}
                        height={650}
                    />
                </ContainerImg>
            </Itemscontainer>
        </SigninContainer>
    );
}

export default withTranslation("signin")(Signin);
