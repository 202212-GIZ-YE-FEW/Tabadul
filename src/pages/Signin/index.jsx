import { serverSideTranslations } from "next-i18next/serverSideTranslations";
import React from "react";

import Footer from "@/components/Footer/Footer";
import Navbar from "@/components/Navbar/Navbar";
import Signin from "@/components/Signin";

function SignIn() {
    return (
        <div>
            <Navbar />
            <Signin />
            <Footer />
        </div>
    );
}

export default SignIn;

export async function getStaticProps({ locale }) {
    return {
        props: {
            ...(await serverSideTranslations(locale, ["signin"])),
            // Will be passed to the page component as props
        },
    };
}
