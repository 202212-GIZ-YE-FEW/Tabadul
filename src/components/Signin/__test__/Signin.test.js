import renderer from "react-test-renderer";
import Signin from "../index";

it("renders correctly", () => {
    const tree = renderer.create(<Signin />).toJSON();
    expect(tree).toMatchSnapshot();
});

jest.mock("next/router", () => ({
    useRouter() {
        return {
            route: "/",
            pathname: "",
            query: "",
            asPath: ",",
        };
    },
}));
