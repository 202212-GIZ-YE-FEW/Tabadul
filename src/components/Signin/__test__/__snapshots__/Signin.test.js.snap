// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-0"
>
  <div
    className="css-132yxzz"
  >
    <div
      className="css-1caouas"
    >
      <h1
        className="css-wr410z"
      >
        Login
      </h1>
      <form
        className="css-0"
        onSubmit={[Function]}
      >
        <div
          className="css-1ohf0ui"
        >
          <div
            className="css-tjuh7q"
          >
            <div
              className="css-crx4p4"
            >
              <label
                className="css-3vajn4"
              >
                Email
              </label>
              <input
                className="css-17aovyj"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Enteryouremail"
                type="text"
                value=""
              />
            </div>
            <div
              className="css-crx4p4"
            >
              <label
                className="css-3vajn4"
              >
                Password
              </label>
              <input
                className="css-17aovyj"
                onChange={[Function]}
                placeholder="Enteryourpassword"
                type="password"
                value=""
              />
            </div>
            <label
              className="css-1xfha8l"
            >
              <input
                checked={false}
                className="css-gpgqoe"
                onChange={[Function]}
                type="checkbox"
              />
              keepmeloggedin
            </label>
            <button
              className="css-otrrrr"
              type="submit"
            >
              <span
                className="css-1wr31v7"
              >
                Loginbutton
              </span>
            </button>
          </div>
        </div>
      </form>
      <div
        className="css-tjuh7q"
      >
        <div
          className="css-1brzwz1"
        >
          <p
            className="css-lx5yh4"
          >
            Dothaveaccount
            <a
              className="css-175it1b"
              href="/Signup"
            >
              Signup
            </a>
          </p>
          <a
            className="css-gw1mbz"
            href="/Forgetpassword"
          >
            ForgetPassword
          </a>
        </div>
        <p
          className="css-uc1awu"
        >
          Loginwith
        </p>
        <div
          className="css-1mzzuk6"
        >
          <img
            alt="Facebook"
            className="css-1vai4q9"
            data-nimg="1"
            decoding="async"
            height={42.3}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/images/Facebook.svg"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={42.3}
          />
          <img
            alt="Instagram"
            className="css-1vai4q9"
            data-nimg="1"
            decoding="async"
            height={42.3}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/images/Instagram.svg"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={42.3}
          />
          <img
            alt="Linkedin"
            className="css-1vai4q9"
            data-nimg="1"
            decoding="async"
            height={42.3}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/images/Linkedin.svg"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={42.3}
          />
        </div>
      </div>
    </div>
    <div
      className="css-12lhnmd"
    >
      <img
        alt="sign in"
        className="css-0"
        data-nimg="1"
        decoding="async"
        height={650}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/images/Signin.svg"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={750}
      />
    </div>
  </div>
</div>
`;
